sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -298,4 +298,28 @@ class DataFrameFunctionsSuite extends QueryTest {
       Seq(Row(2), Row(0), Row(3))
     )
   }
+
+  test("array contains function") {
+    val df = Seq(
+      (Seq[Int](1, 2), "x"),
+      (Seq[Int](), "y"),
+      (null, "z")
+    ).toDF("a", "b")
+    checkAnswer(
+      df.select(array_contains("a", 1)),
+      Seq(Row(true), Row(false), Row(false))
+    )
+    checkAnswer(
+      df.selectExpr("array_contains(a, 1)"),
+      Seq(Row(true), Row(false), Row(false))
+    )
+    // checkAnswer(
+    //   df.select(array_contains("a", null)),
```

Comment:
Hmm, that's a good question, I am wondering if we need to add the implicit casting for the second argument of the `ArrayContains`, that mean we need to inherit from the `ImplicitCastInputTypes`. and keep the first input data type(expected data type) unchanged, but the second input data type depends on the first one.